spec/rails_helper.rb

```diff
@@ -84,6 +84,7 @@
   config.include ViewComponent::SystemTestHelpers, type: :component
   config.include Capybara::RSpecMatchers, type: :component
   config.include Devise::Test::ControllerHelpers, type: :component
+  config.include Devise::Test::ControllerHelpers, type: :helper
```

Comment:
https://github.com/heartcombo/devise#controller-tests looks to say Devise::Test::IntegrationHelpers should be used for newer rails

Reply:
That doesn't work though. The error without this, or with `IntegrationHelpers` has:

```
If you are seeing this on one of your tests, ensure that your tests are either executing the Rails middleware stack or that your tests are using the `Devise::Test::ControllerHelpers` module to inject the `request.env['warden']` object for you.
```

Reply:
my mistake - rspec looks to be Devise::Test::ControllerHelpers only